docs/source/examples/ddd_tutorial.myst

```diff
@@ -0,0 +1,285 @@
+def get_circuit(depth: int, seed: int) -> Tuple[cirq.Circuit, str]:
+    circuit, correct_bitstring = benchmarks.generate_mirror_circuit(
+        nlayers=depth,
+        two_qubit_gate_prob=1.0,
+        connectivity_graph=computer,
+        two_qubit_gate_name="CNOT",
+        seed=seed,
+        return_type="cirq",
+    )
+    return circuit, correct_bitstring
+```
+
+### Helper methods
+
+It will be useful later to us to have a number of utility functions for counting CNOT gates and operations.
+
+```{code-cell} ipython3
+def get_num_cnot_count(circuit: cirq.Circuit) -> int:
+    """Determine number of cnot gates in a given `Circuit` object."""
+    # Count CNOT gates for `cirq`-type circuit objects:
+    num_cnots: int = 0
+    for instruction in circuit.all_operations():
+        if isinstance(instruction, cirq.CXPowGate):
+            num_cnots += 1
+    return num_cnots
+
+
+def get_oneq_count(circuit: cirq.Circuit) -> int:
+    return len(list(circuit.all_operations())) - get_num_cnot_count(circuit)
+```
+
+## Define the executor
+
+Now that we have a circuit, we define the `execute` function which inputs a circuit and returns an expectation value - here, the
+frequency of sampling the correct bitstring.
+
+```{code-cell} ipython3
+def execute(
+    circuits: Union[cirq.Circuit, List[cirq.Circuit]],
+    shots: int,
+    correct_bitstring: str,
+    is_noisy: bool = True,
+) -> List[float]:
+    """Executes the input circuit(s) and returns ⟨A⟩, where
+    A = |correct_bitstring⟩⟨correct_bitstring| for each circuit.
+    """
+    if not isinstance(circuits, list):
+        circuits = [circuits]
+
+    # Store all circuits to run in list to be returned.
+    to_run: List[cirq.Circuit] = []
+
+    # Compile circuits to appropriate gateset.
+    for circuit in circuits:
+        circuit_to_run = circuit.copy()
+        if is_noisy:
+            to_run.append(circuit_to_run.with_noise(cirq.amplitude_damp(0.05)))
+        else:
+            to_run.append(circuit_to_run)
+
+    backend = cirq.Simulator()
+
+    # Process each job.
+    results: List[float] = []
+    for circ_to_run in to_run:
+        counts = 0
+        for _ in range(shots):
+            result = backend.run(
+                circ_to_run + cirq.measure(*sorted(circuit.all_qubits()))
+            )
+            bitstring = list(result.measurements.values())[0][0].tolist()
+            if bitstring == correct_bitstring:
+                counts += 1
+        results.append(counts / shots)
+
+    return results
+```
+
+## Select the DDD sequences to be applied
+We now import a DDD _rule_ from Mitiq, i. e., a function that generates DDD sequences of different length.
+In this example, with simulated amplitude damping noise, we opt for YY sequences (pairs of Pauli Y operations).
+
+```{code-cell} ipython3
+from mitiq import ddd
+
+rule = ddd.rules.yy
+```
+
+## Running the experiment
+
+```{code-cell} ipython3
+:tags: ["skip-execution"]
+cnot_counts, oneq_counts = [], []
```

Comment:
Unsure what these are being used for?

Reply:
Good catch, used for... nothing in this example.